src/vec_int16_ppc.h

```diff
@@ -387,4 +1064,88 @@ vec_srahi (vi16_t vra, const unsigned int shb)
   return (vi16_t) result;
 }
 
+/** \brief Vector Merge Even Halfwords.
+ *
+ * Merge the even halfword elements from the concatenation of 2 x
+ * vectors (vra and vrb).
+ *
+ * \note This function implements the operation of a Vector Merge Even
+ * Halfword instruction, if the PowerISA included such an instruction.
+ * This implementation is NOT endian sensitive and the function is
+ * stable across BE/LE implementations. Using big-endian element
+ * numbering:
+ * - res[0] = vra[0];
+ * - res[1] = vrb[0];
+ * - res[2] = vra[2];
+ * - res[3] = vrb[2];
+ * - res[4] = vra[4];
+ * - res[5] = vrb[4];
+ * - res[6] = vra[6];
+ * - res[7] = vrb[6];
+ *
+ *  |processor|Latency|Throughput|
+ *  |--------:|:-----:|:---------|
+ *  |power8   | 2-13  | 2/cycle  |
+ *  |power9   | 3-14  | 2/cycle  |
+ *
+ * @param vra 128-bit vector unsigned short.
+ * @param vrb 128-bit vector unsigned short.
+ * @return A vector merge from only the even halfwords of vra and vrb.
+ */
+static inline vui16_t
+vec_vmrgeh (vui16_t vra, vui16_t vrb)
+{
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+  const vui16_t permute =
+      { 0x0302,0x1312, 0x0706,0x1716, 0x0B0A,0x1B1A, 0x0F0E,0x1F1E };
+  return vec_perm (vrb, vra, (vui8_t)permute);
+#else
+  const vui16_t permute =
+      { 0x0001,0x1011, 0x0405,0x1415, 0x0809,0x1819, 0x0C0D,0x1C1D };
+  return vec_perm (vra, vrb, (vui8_t)permute);
+#endif
+}
+
+/** \brief Vector Merge Odd Halfwords.
+ *
+ * Merge the odd halfword elements from the concatenation of 2 x
+ * vectors (vra and vrb).
+ *
+ * \note This function implements the operation of a Vector Merge Odd
+ * Halfword instruction, if the PowerISA included such an instruction.
+ * This implementation is NOT endian sensitive and the function is
+ * stable across BE/LE implementations. Using big-endian element
+ * numbering:
+ * - res[0] = vra[1];
+ * - res[1] = vrb[1];
+ * - res[2] = vra[3];
+ * - res[3] = vrb[3];
+ * - res[4] = vra[5];
+ * - res[5] = vrb[5];
+ * - res[6] = vra[7];
+ * - res[7] = vrb[7];
+ *
+ *  |processor|Latency|Throughput|
+ *  |--------:|:-----:|:---------|
+ *  |power8   | 2-13  | 2/cycle  |
+ *  |power9   | 3-14  | 2/cycle  |
+ *
+ * @param vra 128-bit vector unsigned short.
+ * @param vrb 128-bit vector unsigned short.
+ * @return A vector merge from only the odd halfwords of vra and vrb.
+ */
+static inline vui16_t
+vec_vmrgoh (vui16_t vra, vui16_t vrb)
+{
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+  const vui16_t permute =
```

Comment:
Here, like well above, it seems that declaring and initializing `permute` as `vui8_t` would be easier to read. (4 lines below as well.)

Reply:
See reply for vec_vmrgeh above. This is hard enough and these are the tricks I use.

No plan to change this.